reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -100,4 +100,35 @@ bool BftReconfigurationHandler::verifySignature(const std::string& data, const s
   if (pub_key_ == nullptr && verifier_ == nullptr) return false;
   return pub_key_->verify(data, signature) || verifier_->verify(data, signature);
 }
+
+bool ReconfigurationHandler::handle(const UnwedgeCommand& cmd,
```

Comment:
This is the tricky part.
The command handler is being called **after** consensus. However, on wedge, the consensus itself is blocked.
This is why we need to verify this request when getting it, prior to consensus.
On the primary replica this will on **onMessage<ClientRequestMsg>** (see https://github.com/vmware/concord-bft/blob/master/bftengine/src/bftengine/ReplicaImp.cpp#L223).
On the follower replica, this would be when they check the conditions to perPrepare (see https://github.com/vmware/concord-bft/blob/master/bftengine/src/bftengine/ReplicaImp.cpp#L668).
We need to define a special case for these two spots in which, if we get an unwedge command and it is valid, it can pass consensus.
Once this is done, the handler for unwedge is simply to clear the wedge checkpoint (no need to verify the signatures again)

Reply:
After discussing with @ittaiab, we decided to leave it as it is under the assumption that we will have n/n replies for the unwedge status.
Once the work on epochs will be done (see https://github.com/vmware/concord-bft/pull/1608) the unwedge read request will cause the replica to start a new epoch. Eventually, wedge will be tied to an epoch.
Meaning, when issuing a wedge, it means that wedge all write messages until we start a new epoch